cns/restserver/api.go

```diff
@@ -899,6 +902,99 @@ func (service *HTTPRestService) getNetworkContainerByOrchestratorContext(w http.
 	logger.Response(service.Name, getNetworkContainerResponse, returnCode, err)
 }
 
+// getOrRefreshNetworkContainers is to check whether refresh association is needed.
+// If received  "GET": return all NCs in CNS's state file to DNC for follow-up check
+// If received "POST": Store all the NCs (from the request body that client sent) into CNS's state file
+func (service *HTTPRestService) getOrRefreshNetworkContainers(w http.ResponseWriter, r *http.Request) {
+	logger.Printf("[Azure CNS] getOrRefreshNetworkContainers")
+
+	switch r.Method {
+	case "GET":
+		// return all NCs in CNS
+		networkContainersDetails := make(map[string]cns.ContainerStatus)
+		for ncID := range service.state.ContainerStatus {
```

Comment:
does this not need a read lock?

Reply:
Yes, add a read lock would be more accurate.